### ISSUE TITLE
[Feature Request] false positive on consider-using-enumerate

### ISSUE THREAD

**Issue by nlgranger** [PyCQA/pylint#2477]
### Is your feature request related to a problem? Please describe

Pylint suggests using enumerate instead of `range(len(obj))`.
This solution is impossible if `obj` does not support `__iter__`, in particular if we are implementing it:

```python
    def __iter__(self):
        for i in range(len(self)):
            yield self[i]
```

### Describe the solution you'd like

add a special case for `range(len(self))` inside `__iter__` method.

This admittedly is a very low priority request, but I'd reckon this is not a rare situation, I'm not sure if you like to take these kind of cases into account, feel free to close.

**Comment by PCManticore** [comment]
Yes, we can probably improve this check to not account for this case.
